lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -242,7 +242,7 @@ public function fetch($fetchMode = null)
     {
         $values = $this->_fetch();
         if (null === $values) {
-            return null;
```

Comment:
What happens when a `null` is actually retrieved? Is it supposed to always retrieve an `array`?